### PR TITLE
Haze mask fix

### PR DESCRIPTION
**Pull request by NissimHadar**
Replaces the stencil mask used when the haze effect has been enabled.  The previous masked caused a problem when looking at the black mask in the preview, or in the HMD when looking to the side.

## TEST PLAN
1) Menu->Display->Oculus Rift to checked
2) Menu->Display->Disable Preview to unchecked
3) Menu->Display->Mono Preview to unchecked
4) Verify mask around both images is black
5) Menu->Display->Desktop to checked
6) Activate Haze
7)) Menu->Display->Oculus Rift to checked 
8) Verify mask around both images is still black

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4619/HighFidelity-Beta-PR11732-ad11606e34e0cbc9b5c124b92aeba6d76398f47c.dmg

File: libraries/render-utils/src/StencilMaskPass.cpp [hunk 1]
```diff
@@ -128,6 +128,11 @@ void PrepareStencil::testNoAA(gpu::State& state) {
         gpu::State::STENCIL_OP_KEEP, gpu::State::STENCIL_OP_KEEP, gpu::State::STENCIL_OP_KEEP));
 }
 
+void PrepareStencil::testMaskNoAA(gpu::State& state) {
```
Comment (samcake) [inline libraries/render-utils/src/StencilMaskPass.cpp]: Can you add a comment similar to the other function to explain precisely what this function does.
thanks

**Comment by samcake** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4622/HighFidelity-Beta-PR11732-ad11606e34e0cbc9b5c124b92aeba6d76398f47c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4622/HighFidelity-Beta-PR11732-ad11606e34e0cbc9b5c124b92aeba6d76398f47c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4626/HighFidelity-Beta-PR11732-2e6c849b63f55f453c08a9da64271dfcbbc1857c.dmg

**Comment by cain-kilgore** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4629/HighFidelity-Beta-PR11732-2e6c849b63f55f453c08a9da64271dfcbbc1857c.dmg

**Comment by samcake** [comment]
build this please

**Comment by cain-kilgore** [comment]
@samcake Is there a test plan for this?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4634/HighFidelity-Beta-PR11732-2e6c849b63f55f453c08a9da64271dfcbbc1857c.dmg

**Comment by NissimHadar** [comment]
Please build this

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4653/HighFidelity-Beta-PR11732-d8191e3bd2122f8e45f1ce70d2c4fce98a9f5503.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4653/HighFidelity-Beta-PR11732-d8191e3bd2122f8e45f1ce70d2c4fce98a9f5503.exe

**Comment by Nex-Pro** [comment]
@samcake @SamGondelman Will the haze show in the zone stack ( debugDeferred )

**Comment by samcake** [comment]
@Nex-Pro at some point yes, i didn't have time to look at it yet

**Comment by Nex-Pro** [comment]
Testing

**Comment by Nex-Pro** [comment]
### PASS

All good. Mask remains black when haze on the Zone is enabled while observing with mono preview off.


capture

**Comment by Nex-Pro** [comment]
@NissimHadar ^

**Pull request merged**